darts/models/forecasting/regression_model.py

```diff
@@ -170,94 +192,170 @@ def encode_year(idx):
+                            and isinstance(comp_lags[1], int),
+                            f"`{lags_name}` tuple must be of length 2, and must contain two integers",
+                            logger,
+                        )
+
+                        raise_if(
+                            isinstance(comp_lags[0], bool)
+                            or isinstance(comp_lags[1], bool),
+                            f"`{lags_name}` tuple must contain integers, not bool",
+                            logger,
+                        )
+
+                        raise_if_not(
+                            comp_lags[0] >= 0 and comp_lags[1] >= 0,
+                            f"`{lags_name}` tuple must contain positive integers. Given: {comp_lags}.",
+                        )
+                        raise_if(
+                            comp_lags[0] == 0 and comp_lags[1] == 0,
+                            f"`{lags_name}` tuple cannot be (0, 0) as it corresponds to an empty list of lags.",
+                            logger,
+                        )
+                        tmp_components_lags[comp_name] = list(
+                            range(-comp_lags[0], comp_lags[1])
+                        )
+                    elif isinstance(comp_lags, list):
+                        for lag in comp_lags:
+                            raise_if(
+                                not isinstance(lag, int) or isinstance(lag, bool),
+                                f"`{lags_name}` list must contain only integers. Given: {comp_lags}.",
+                            )
+                        tmp_components_lags[comp_name] = sorted(comp_lags)
+                    else:
+                        invalid_type = True
+                        supported_types = "tuple or a list"
+                else:
+                    if isinstance(comp_lags, int):
+                        raise_if_not(
+                            comp_lags > 0,
+                            f"`{lags_name}` integer must be strictly positive . Given: {comp_lags}.",
+                        )
+                        tmp_components_lags[comp_name] = list(range(-comp_lags, 0))
+                    elif isinstance(comp_lags, list):
+                        for lag in comp_lags:
+                            raise_if(
+                                not isinstance(lag, int) or (lag >= 0),
+                                f"`{lags_name}` list must contain only strictly negative integers. Given: {comp_lags}.",
+                            )
+                        tmp_components_lags[comp_name] = sorted(comp_lags)
+                    else:
+                        invalid_type = True
+                        supported_types = "strictly positive integer or a list"
+
+                if invalid_type:
+                    raise_log(
+                        ValueError(
+                            f"When passed in a dictionary, `{lags_name}` for component {comp_name} must be either a "
```

Comment:
related to my previous comment
```suggestion
                            f"`{lags_name}` - `{comp_name}`: must either be a "
```